ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -144,17 +146,31 @@ private OperationController getOperationController() {
       throw new IllegalArgumentException("blobId or options must not be null");
     }
     currentOperationsCount.incrementAndGet();
+    final FutureResult<GetBlobResult> futureResult = new FutureResult<>();
```

Comment:
why was the location of this line and the next moved?

Reply:
if BlobId creation throws an exception, we need an handle to futureResult and GetBlobOptionsInternal and hence